Statically link binaries and remove debug information

pjbgf opened the pull request:
Statically link the generated binaries as to avoid issues when the application is linked to a different GLIBC version than the one available when `kube-bench` is used.
This also removes the debug information, resulting on the assets being half the current size (`~13MB` each instead of `~25MB`).

Fixes https://github.com/aquasecurity/kube-bench/issues/1592.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/aquasecurity/kube-bench?pullRequest=1615) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/aquasecurity/kube-bench?pullRequest=1615) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/aquasecurity/kube-bench?pullRequest=1615) it.</sub>

pjbgf commented:
The lint error is orthogonal to the changes:

```
$ golangci-lint run
WARN The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
WARN The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
ERRO [linters_context] deadcode: This linter is fully inactivated: it will not produce any reports.
ERRO [linters_context] varcheck: This linter is fully inactivated: it will not produce any reports.
```

Should I remove the two deprecated linters as part of this PR?

chen-keinan commented:
@pjbgf please rebase with upstream to fix the lint issue.

mozillazg reviewed (approved):
LGTM, Thanks for your contribution!
